spec/factories/report.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+def test_result_score(counter, threshold)
+  threshold + (counter.zero? ? -1 : 1) * (0..(100 - threshold)).to_a.sample
+end
+
+FactoryBot.define do
+  factory :v2_report, class: 'V2::Report' do
+    to_create do |instance, context|
+      policy = FactoryBot.create(
+        :v2_policy,
+        :for_tailoring,
+        title: context.title,
+        system_count: context.assigned_system_count,
+        os_major_version: context.os_major_version,
+        compliance_threshold: context.compliance_threshold,
+        business_objective: context.business_objective,
+        account: context.account,
+        supports_minors: context.supports_minors
+      )
+
+      supported_counter = policy.systems.count - context.unsupported_system_count
+      compliant_counter = context.compliant_system_count
```

Comment:
You have 2 numbers decreasing in an iteration which is quite hard to follow and debug. What about splitting up the iteration into multiple ones and running it with traits?